editoast/src/views/timetable.rs

```diff
@@ -126,21 +126,41 @@ async fn get(
     State(db_pool): State<DbConnectionPoolV2>,
     Extension(auth): AuthenticationExt,
     Path(TimetableIdParam { id: timetable_id }): Path<TimetableIdParam>,
-) -> Result<Json<TimetableDetailedResult>> {
+    Query(pagination_params): Query<PaginationQueryParams>,
+) -> Result<Json<ListTrainSchedulesResponse>>  {
     let authorized = auth
-        .check_roles([BuiltinRole::TimetableRead].into())
-        .await
-        .map_err(AuthorizationError::AuthError)?;
+    .check_roles([BuiltinRole::TimetableRead].into())
+    .await
+    .map_err(AuthorizationError::AuthError)?;
     if !authorized {
         return Err(AuthorizationError::Forbidden.into());
     }
 
     let conn = &mut db_pool.get().await?;
-    let timetable = TimetableWithTrains::retrieve_or_fail(conn, timetable_id, || {
+    Timetable::retrieve_or_fail(conn, timetable_id, || {
         TimetableError::NotFound { timetable_id }
     })
     .await?;
-    Ok(Json(timetable.into()))
+
+    let settings = pagination_params
+    .validate(1000)?
+    .warn_page_size(100)
```

Comment:
Let's enforce pagination for new endpoints. What about

```suggestion
    .validate(25)?
```

?